data/multiplayer/scenarios/4p_King_of_the_Hill.cfg

```diff
@@ -54,9 +48,167 @@
+        [slider]
+            id="bonus_income"
+            default=15
+            min=5
+            max=40
+            name= _ "Hill Bonus Income"
+            description= _ "The amount of extra gold per turn granted to the leader who occupies the hill keep."
+        [/slider]
+        [slider]
+            id="chest_gold"
+            default=100
+            min=25
+            max=500
+            step=25
+            name= _ "Treasure Chest Gold"
+            description= _ "The amount of gold stashed inside the treasure chest on the hill."
+        [/slider]
+    [/options]
+
+    [event]
+        name=prestart
+        [objectives]
+            [objective]
+                description= _ "Defeat enemy leaders"
+                condition=win
+            [/objective]
+            [note]
+                description= _ "A side with a leader holding the hill keep gets +$bonus_income gold income each turn."
+            [/note]
+            [note]
+                description= _ "This side will also get a second leader on its original keep."
+            [/note]
+            [note]
+                description= _ "The treasure chest on the hill keep can only be unlocked by a leader, for a bonus of $chest_gold gold."
+            [/note]
+        [/objectives]
+        # The hill is always visible to all players.
+        [lift_fog]
+            x,y=18,18
+            radius=4
+            multiturn=yes
+        [/lift_fog]
+        [remove_shroud]
+            x,y=18,18
+            radius=4
+        [/remove_shroud]
+    [/event]
+
+    # A leader on the hill keep gets a kingly tribute each turn.
+    [event]
+        name=turn refresh
+        first_time_only=no
+        [if]
+            [have_unit]
+                x,y=18,18
+                canrecruit=yes
+                side=$side_number
+            [/have_unit]
+            [then]
+                [gold]
+                    side=$side_number
+                    amount=$bonus_income
+                [/gold]
+                [scroll_to]
+                    x,y=18,18
+                [/scroll_to]
+                [floating_text]
+                    x,y=18,18
+                    text= _ "<span color='gold'>+$bonus_income gold</span>"
+                [/floating_text]
+                # And it gets a second leader (appearing on its original keep), if it does not already have a second leader.
+                [if]
+                    [have_unit]
+                        canrecruit=yes
+                        side=$side_number
+                        count=1
+                    [/have_unit]
+                    [then]
+                        [store_unit]
+                            [filter]
+                                x,y=18,18
+                            [/filter]
+                            variable=twin
+                        [/store_unit]
+                        [switch]
+                            variable=side_number
+                            [case]
```

Comment:
Using [store_starting_location] would probably be better here.

Reply:
That is much cleaner, thanks!